## Step 6 — Basic Quality Scan

In [5]:
import pandas as pd

# Load dataset (ubah path sesuai lokasi file Anda)
file_path = r"C:\Users\Michael Marcel\OneDrive\Desktop\TA\Data\Dataset\student_lifestyle_100k.csv"

df = pd.read_csv(file_path)

df.head()

,Student_ID,Age,Gender,Department,CGPA,Sleep_Duration,Study_Hours,Social_Media_Hours,Physical_Activity,Stress_Level,Depression
0,1001,22,Female,Science,3.50,7.3,3.3,3.4,114,5,False
1,1002,20,Male,Engineering,2.72,5.5,7.2,6.0,142,2,False
2,1003,20,Male,Medical,3.01,5.4,2.3,1.8,137,3,False
3,1004,21,Male,Engineering,3.63,8.1,2.0,4.6,130,3,False
4,1005,19,Male,Arts,3.14,6.8,2.6,4.3,4,6,False


In [6]:
# Step 6.1 — Missing Values per Column
missing_values = df.isnull().sum()

print('Jumlah Missing Values per Kolom:')
missing_values

Jumlah Missing Values per Kolom:


Student_ID            0
Age                   0
Gender                0
Department            0
CGPA                  0
Sleep_Duration        0
Study_Hours           0
Social_Media_Hours    0
Physical_Activity     0
Stress_Level          0
Depression            0
dtype: int64

In [7]:
# Persentase Missing Values
missing_percent = (df.isnull().sum() / len(df)) * 100

missing_percent

Student_ID            0.0
Age                   0.0
Gender                0.0
Department            0.0
CGPA                  0.0
Sleep_Duration        0.0
Study_Hours           0.0
Social_Media_Hours    0.0
Physical_Activity     0.0
Stress_Level          0.0
Depression            0.0
dtype: float64

In [8]:
# Unique Value Count per Column
unique_counts = df.nunique()

print('Jumlah Nilai Unik per Kolom:')
unique_counts

Jumlah Nilai Unik per Kolom:


Student_ID            100000
Age                        7
Gender                     2
Department                 5
CGPA                     237
Sleep_Duration            91
Study_Hours              126
Social_Media_Hours        97
Physical_Activity        150
Stress_Level               9
Depression                 2
dtype: int64

In [9]:
# Unique values khusus kolom kategorikal
categorical_cols = df.select_dtypes(include='object').columns

for col in categorical_cols:
    print(f'\n{col} : {df[col].nunique()} unique values')
    print(df[col].value_counts().head())


Gender : 2 unique values
Gender
Male      50120
Female    49880
Name: count, dtype: int64

Department : 5 unique values
Department
Science        20071
Engineering    20057
Arts           19998
Medical        19964
Business       19910
Name: count, dtype: int64


C:\Users\Michael Marcel\AppData\Local\Temp\ipykernel_19784\2366293200.py:2: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_cols = df.select_dtypes(include='object').columns


In [10]:
# Step 6.3 — Range Nilai Numerik
numeric_cols = df.select_dtypes(include=['int64','float64']).columns

for col in numeric_cols:
    print(f'\n{col}')
    print('Min :', df[col].min())
    print('Max :', df[col].max())


Student_ID
Min : 1001
Max : 101000

Age
Min : 18
Max : 24

CGPA
Min : 1.56
Max : 4.0

Sleep_Duration
Min : 3.0
Max : 12.0

Study_Hours
Min : 0.0
Max : 12.8

Social_Media_Hours
Min : 0.0
Max : 10.0

Physical_Activity
Min : 0
Max : 149

Stress_Level
Min : 2
Max : 10


## Initial Observations

Tuliskan minimal 3 temuan awal, contoh:

1. Berdasarkan hasil pengecekan missing values, seluruh kolom pada dataset memiliki nilai kosong sebesar 0 atau 0%. Hal ini menunjukkan bahwa dataset berada dalam kondisi lengkap (complete data) sehingga tidak diperlukan proses penanganan missing values seperti imputasi maupun penghapusan baris data. Kondisi ini mendukung keandalan analisis lanjutan karena tidak terdapat potensi distorsi akibat data hilang.
2. Rentang usia mahasiswa berada pada interval 18 hingga 24 tahun, yang merepresentasikan kelompok usia tipikal mahasiswa aktif. Nilai IPK (CGPA) memiliki rentang dari 1.56 hingga 4.00 yang menunjukkan variasi performa akademik dari kategori rendah hingga sangat baik. Variasi ini memungkinkan analisis hubungan akademik dengan faktor gaya hidup dan kesehatan mental.
3. Durasi tidur mahasiswa berkisar antara 3 hingga 12 jam per hari, sementara waktu belajar mencapai maksimum 12.8 jam dan penggunaan media sosial hingga 10 jam per hari. Rentang yang cukup lebar ini mengindikasikan adanya heterogenitas pola hidup mahasiswa, yang berpotensi berpengaruh terhadap tingkat stres maupun depresi.